src/gretel_trainer/benchmark/comparison.py

```diff
@@ -89,14 +112,20 @@ def __init__(
         self.datasets = [
             _make_dataset(dataset, self.config.working_dir) for dataset in datasets
         ]
-        self._gretel_executors: Dict[str, Executor] = {}
-        self._custom_executors: Dict[str, Executor] = {}
+        self._gretel_executors: Dict[Tuple[str, str], Executor] = {}
+        self._custom_executors: Dict[Tuple[str, str], Executor] = {}
         self.trainer_project_names: Dict[str, str] = {}
         self.thread_pool = ThreadPoolExecutor(5)
-        self.futures: Dict[str, Future] = {}
+        self.futures: Dict[FutureKeyT, Future] = {}
+
+        self._extra_report_scores = {
+            score_name: ALL_REPORT_SCORES[score_name]
+            for score_name in (additional_report_scores or [])
+            if score_name != "SQS"
+        }
```

Comment:
I felt "extra" was cleaner because I don't see much of a valid reason of dropping `SQS` unless you want no scores at all. But that then makes the semantics of the parameters really confusing thanks to Python not (properly) supporting list-typed parameter defaults. You'd have to have something like `all_report_scores: Optional[List[str]]` where the default is `None` and means `["SQS"]` whereas `[]` means "no scores".

Reply:
Oh, to clarify: I think "extra" makes sense at the public interface. I mean that internally we could fold SQS into the list of extra report scores and just treat them all equivalently once the instance has been initialized (e.g. in the results dict)

Reply:
Makes sense, will implement